sos/policies/runtimes/lxd.py

```diff
@@ -0,0 +1,148 @@
+class LxdContainerRuntime(ContainerRuntime):
+    """Runtime class to use for systems running LXD"""
+
+    name = 'lxd'
+    binary = 'lxc'
+
+    def check_is_active(self, sysroot=None):
+        # the daemon must be running
+        if (is_executable('lxc', sysroot) and
+                (self.policy.init_system.is_running('lxd') or
+                 self.policy.init_system.is_running('snap.lxd.daemon'))):
+            self.active = True
+            return True
+        return False
+
+    def get_containers(self, get_all=False):
+        """Get a list of containers present on the system.
+
+        :param get_all: If set, include stopped containers as well
+        :type get_all: ``bool``
+        """
+        containers = []
+
+        _cmd = "%s list --format json" % (self.binary)
+        if self.active:
+            out = sos_get_command_output(_cmd, chroot=self.policy.sysroot)
+
+            if out["status"] == 0:
+                out_json = json.loads(out["output"])
+
+                for container in out_json:
+                    if container['status'] == 'Running' or get_all:
+                        # takes the form (container_id, container_name)
+                        containers.append(
+                            (container['expanded_config']['volatile.uuid'],
+                             container['name']))
+
+        return containers
+
+    def get_images(self):
+        """Get a list of images present on the system
+
+        :returns: A list of 2-tuples containing (image_name, image_id)
+        :rtype: ``list``
+        """
+        images = []
+        if self.active:
+            out = sos_get_command_output(
+                f"{self.binary} image list --format json",
+                chroot=self.policy.sysroot
+            )
+            if out['status'] == 0:
+                out_json = json.loads(out["output"])
+                for ent in out_json:
+                    # takes the form (image_name, image_id)
+                    images.append((
+                        ent['update_source']['alias'],
+                        ent['fingerprint']))
+        return images
+
+    def get_volumes(self):
+        """Get a list of container volumes present on the system
+
+        :returns: A list of volume IDs on the system
+        :rtype: ``list``
+        """
+        vols = []
+        if self.active:
+
+            # first get the default storage pool
+            out = sos_get_command_output(
+                f"{self.binary} profile list --format json",
+                chroot=self.policy.sysroot
+            )
+            if out['status'] == 0:
+                out_json = json.loads(out['output'])
+                for profile in out_json:
+                    if profile['name'] == 'default':
+                        stg_pool = profile['devices']['root']['pool']
+                        break
+
+            out = sos_get_command_output(
+                f"{self.binary} storage volume list {stg_pool} --format json",
```

Comment:
We should have a default value for `stg_pool`, or raise an exception after the check if we don't find one. Like CodeQL mentions, there's a possibility here for `stg_pool` to be undefined when we make this call to `sos_get_command_output()`

Reply:
Indeed, our CI tests even detected it :) https://cirrus-ci.com/task/5971844470145024